File: Mandatory/003 - Anagram/anagram.py
def anagram(s):
  arrayEvalChars = list(s)
  lengthString = len(arrayEvalChars)
  
  count = -1
  if lengthString % 2 == 0:
    s1 = arrayEvalChars[:(lengthString // 2)]
    s2 = arrayEvalChars[(lengthString // 2):]
    
    for char in s1:      
      try:
        indexOfChar = s2.index(char)
      except ValueError:
        indexOfChar = -1
      
      if indexOfChar > -1:
        s2.remove(char)
  
    count = len(s2)
    
  return count

print(anagram('xaxbbbxx'))

File: Mandatory/003 - Anagram/anagram.rb
def anagram(s)
  arrayEvalChars = s.split('')
  lengthString = arrayEvalChars.length

  count = -1
  if lengthString.even?    
    s1 = arrayEvalChars[0..(-1 + lengthString / 2)]
    s2 = arrayEvalChars[(lengthString / 2).. -1]

    s1.each { |char|
      indexOfChar = s2.index(char)
      s2.delete_at(indexOfChar) if !indexOfChar.nil?
    }

    count = s2.length
  end

  count
end

p anagram('xaxbbbxx')
